### PR TITLE
[CALCITE-3833] Support SemiJoin in EnumerableMergeJoin

### PR DESCRIPTION
**Pull request by rubenada**
Jira: [CALCITE-3833](https://issues.apache.org/jira/browse/CALCITE-3833)

### CODE REVIEW

File: linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java [hunk 1]
```diff
@@ -1971,8 +1973,42 @@ private void closeInner() {
...
+    }
+    if (generateNullsOnRight) {
+      throw new UnsupportedOperationException(
+          "not implemented, mergeJoin with generateNullsOnRight");
+    }
+    return mergeJoin(outer, inner, outerKeySelector, innerKeySelector, null, resultSelector,
+        JoinType.INNER, null);
+  }
+
+  public static boolean isMergeJoinSupported(JoinType joinType) {
```
Comment (hsyuan) [inline linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1988]: Can we add a note that this method is subject to change or remove without notice?

Comment (rubenada) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1988]: Sure, I can add that; but can you please elaborate why do think it is necessary to mention it?

Comment (hsyuan) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1988]: Once all the join type are supported, there is no need for the method.

Comment (rubenada) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1988]: That is an optimistic point of view :) 
Ok, I'll add the note.

Comment (hsyuan) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1988]: :) Indeed, it is up to you.

Comment (rubenada) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1992]: done

**Comment by DonnyZone** [comment]
LGTM!

**Comment by rubenada** [comment]
@hsyuan I am ready to merge this patch. Do you think it would impact [PR#1860](https://github.com/apache/calcite/pull/1860)? What is the current status on that PR, do you expect to merge it soon? Should I wait for it or you think I can go on and merge the current one?

**Comment by hsyuan** [comment]
@rubenada I am ready to merge #1860. But you can go ahead and merge your PR.

**Comment by rubenada** [comment]
Thanks @hsyuan . I will proceed and merge this PR.

**Pull request merged**